--- a/cmake/ZSTDCodec.cmake
+++ b/cmake/ZSTDCodec.cmake
@@ -32,9 +32,11 @@
 find_package(ZSTD)

 if(ZSTD_FOUND)
-    if(TARGET zstd::libzstd_shared)
+    if(TARGET ZSTD::ZSTD)
+        # Target already exists, nothing to do
+    elseif(TARGET zstd::libzstd_shared)
         add_library(ZSTD::ZSTD ALIAS zstd::libzstd_shared)
         set(ZSTD_HAVE_DECOMPRESS_STREAM ON)
     elseif(TARGET zstd::libzstd_static)
         add_library(ZSTD::ZSTD ALIAS zstd::libzstd_static)
         set(ZSTD_HAVE_DECOMPRESS_STREAM ON)
